docs/architecture/front-end-component-structure.md

```diff
@@ -0,0 +1,179 @@
+
+- The entrypoint file for each React component has the same name as the directory, e.g. src/Home/Home.tsx. This is less kind on the import statements (repeats ‘Home’), though we are avoiding many index.ts(x) files, which is problematic for editor tabs and searching.
+
+- Files can only import the entrypoint file from each component directory (it is treated as an index file).
+
+- An exception - sometimes it is useful to group similar components. Where there is no single entrypoint, an index.ts file is useful for single line imports and each component should have its own unit test file, see for example the ErrorPages/ below.
+
+- Styles are implemented in TypeScript inside their component files and sass files (scss/\*.scss) are removed in line with [#239](https://github.com/mitodl/mit-open/issues/239).
+
+## Shared workspaces
+
+Workspaces that can be shared across application root workspaces can follow the same naming for categorization, with an ol- prefix, providing us:
+
+- `ol-page-components` - sub page components that may be routable (React Router nesting) and/or are areas of functionality that form a section of the page and incorporate many components, an in-page comments board, for example. They have knowledge of the application and can import up outside their workspace e.g. to use any ol-components or the api service.
+
+- `ol-components` - a catalog of React presentational components, forming our OL component library. Components can import sideways, but cannot import outside of their workspace. They cannot therefore interact directly with app level services or functionality except where passed to them as functions (handler callbacks, etc).
+
+- `ol-services` - we may want to move any bridges to backend services for reuse in other applications.
+
+- `ol-utilities` - shared utility code.
+
+- `ol-common` - For code common to `mit-open`, `ol-page-components` and `ol-components`. Used for TypeScript definitions and anything that has visibility of the app or is not a utility or service.
+
+## Large shared components
+
+Some components are large, contain a lot of functionality or can be considered a “mini app” for embedding within root applications. It may not necessarily be only their size that qualifies them for different treatment - perhaps they have special build chains, are not built on React, or have other requirements. The example in the project is the CKEditor, a rich text editor that has its own workspace, ol-ckeditor, in which we wrap and configure it for use in our application.
+
+These will always have their own package.json (as a defining factor), so are publishable independently to a package repository and should be, for reuse outside the project. To contain these and prevent our top level directories from sprawling, let’s introduce an ol-packages workspace that constitutes our npm packages library. Each of its child directories may contain a workspace of its own.
```

Comment:
All the workspaces have their own `package.json`, so all are packages in this sense. Not all are publishable, though.

>  let’s introduce an ol-packages work...

IMO, it's a bit awkward for this file to talk about things we should do. If it's committed, I would expect it to be things we have done.

Reply:
Agree, removed.